test/CompressedRelationsTest.cpp

```diff
@@ -31,8 +31,30 @@ Id V(int64_t index) {
 auto getBlockMetadataRangesfromVec =
     [](const std::vector<CompressedBlockMetadata>& blockMetadata)
     -> BlockMetadataRanges {
-  BlockMetadataSpan blockSpan(blockMetadata);
-  return {{blockSpan.begin(), blockSpan.end()}};
+  BlockMetadataSpan blockMetadataSpan(blockMetadata);
+  const size_t numBlocks = blockMetadataSpan.size();
+  if (numBlocks < 3) {
+    return {{blockMetadataSpan.begin(), blockMetadataSpan.end()}};
+  }
+  // If the BlockMetadataSpan contains more than three block values, split it
+  // into three random subspans.
+  BlockMetadataRanges ranges;
+  auto begin = blockMetadataSpan.data();
+  ad_utility::FastRandomIntGenerator<size_t> gen;
+  const size_t minNumBlocksSplit = 1;
+  size_t remainingBlocksForSplit = numBlocks - 3 * minNumBlocksSplit;
+  // Split 1
+  size_t offset = gen() % (remainingBlocksForSplit + 1);
+  size_t split1 = minNumBlocksSplit + offset;
+  remainingBlocksForSplit = numBlocks - split1 - 2 * minNumBlocksSplit;
+  // Split 2
+  offset = gen() % (remainingBlocksForSplit + 1);
+  size_t split2 = split1 + minNumBlocksSplit + offset;
```

Comment:
The splitting logic looks really complicated, 
Why don't you just sample two random splitting points independently in [0, size())`, order and apply them.
Sure, sometimes we will have empty subblocks, but that should still work, shouldn't it?
Or does your code currently break if some of the blocksubranges are empty?